### PR TITLE
Adding junit reports to test and test-integration

### LINKED ISSUE #137: Test grid should show all the tests successful or failure
https://k8s-testgrid.appspot.com/sig-apps#pull-jobset-test-integration-main only shows a single row which is not ideal.

We enabled this in Kueue and it has helped identify flaky tests.

I think uploading junit reports for each test was all we needed for support in Kueue.

### PR DESCRIPTION
**Pull request by clearnicki**
This PR fixes #137 

For test and test-integration - I added junit reports, as well as making sure gotestsum will be downloaded.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -223,7 +226,8 @@ kind-image-build: kind image-build
 .PHONY: test-integration
 test-integration: manifests generate fmt vet envtest ginkgo ## Run tests.
 	KUBEBUILDER_ASSETS="$(shell $(ENVTEST) use $(ENVTEST_K8S_VERSION) -p path)" \
-	$(GINKGO) -v $(INTEGRATION_TARGET)
+	$(GINKGO) --junit-report=junit.xml --output-dir=$(ARTIFACTS) -v $(INTEGRATION_TARGET)
+
 
 .PHONY: test-e2e-kind
 test-e2e-kind: manifests generate kustomize fmt vet envtest ginkgo kind-image-build
```
[the remark this answers is not in the document]
Comment (kannon92) [reply Makefile:233]: I think you can add `--junit-report=junit.xml` to https://github.com/kubernetes-sigs/jobset/blob/main/hack/e2e-test.sh#L30 or at least something similar to get our e2e tests to report the junit also.

[the remark this answers is not in the document]
Comment (clearnicki) [reply Makefile:233]: ok - I can work on that. thanks!

**Comment by kannon92** [comment]
@clearnicki Thank you for this PR.  

For @danielvegamyhre, https://testgrid.k8s.io/sig-apps#pull-jobset-test-integration-main and https://testgrid.k8s.io/sig-apps#pull-jobset-test-unit-main look very good!  Just need the e2e tests and I think this PR is good to go.

**Pull request merged**